Makefile

```diff
@@ -188,7 +188,7 @@ kernel/sys/version.o: kernel/*/*.c kernel/*.c
 
 hdd/mod/%.ko: modules/%.c ${HEADERS}
 	@${BEG} "CC" "$< [module]"
-	@${CC} -T modules/link.ld -I./kernel/include -nostdlib ${CFLAGS} -c -o $@ $< ${ERRORS}
+	@${CC} -T modules/link.ld -I./kernel/include -I./userspace -nostdlib ${CFLAGS} -c -o $@ $< ${ERRORS}
```

Comment:
Plop your system headers in `userspace/lib` and then symlink them in `kernel/include/mod`, rather than having the modules pull in the include path. Keeps things nice and separate.